pkg/templates/compile.go

```diff
@@ -87,64 +87,64 @@ func Parse(filePath string, preprocessor Preprocessor, options *protocols.Execut
 			// as not using a cache at all, but may be necessary.
 
 			for i, r := range tplCopy.RequestsDNS {
-				rCopy := *r
```

Comment:
ig this copy of `r` was intentional, because without copy it would modify the reference potentially leading to unintended consequences for `UpdateOptions` we do later.

Earlier we were using direct `r` pointer but it was changed here: https://github.com/projectdiscovery/nuclei/commit/5b89811b906f086e47fdf5357a5ee367f8f1daef#diff-5ff00a91c11854bc967fe7a6dfcee1f0616ee8f3e06ad9715d62e1a848c1b99dR80-R90

Reply:
this is very expensive, i can see currently we can use safely but seems HDM has added this for make multi instance caching safe.reverting to HDM changes, but its continue to have cpu performance degrade until we start using those.